Makefile

```diff
@@ -1,107 +1,121 @@
-	@$(PY) $(BUILD)                                                        \
-		--input=$<                                                     \
-		--output=$@                                                    \
-		--version=$(VERSION)                                           \
-		$(if $(call nofea,$@),,--feature-file=$(GSUB))                 \
-		;
+$(BUILDDIR):
+	mkdir -p $@
+
+$(BUILDDIR)/%.otl.otf: $(SOURCEDIR)/%.sfd $(GSUB) $(BUILD) | $(BUILDDIR)
+	$(info       BUILD  $(*F))
+	$(PY) $(BUILD) \
+		--input=$< \
+		--output=$@ \
+		--version=$(VERSION) \
+		$(if $(call nofea,$@),,--feature-file=$(GSUB))
 
 $(BUILDDIR)/%.hint.otf: $(BUILDDIR)/%.otl.otf
-	@echo "       HINT  $(*F)"
-	@rm -rf $@.log
-	@psautohint $< -o $@ --log $@.log
+	$(info        HINT  $(*F))
+	rm -rf $@.log
+	psautohint $< -o $@ --log $@.log
 
 $(BUILDDIR)/%.subr.otf: $(BUILDDIR)/%.hint.otf
-	@echo "       SUBR  $(*F)"
-	@tx -cff +S +b $< $(@D)/$(*F).cff 2>/dev/null
-	@sfntedit -a CFF=$(@D)/$(*F).cff $< $@
+	$(info        SUBR  $(*F))
+	tx -cff +S +b $< $(@D)/$(*F).cff 2> /dev/null
+	sfntedit -a CFF=$(@D)/$(*F).cff $< $@
 
 %.otf: $(BUILDDIR)/%.subr.otf
-	@cp $< $@
-
-$(BUILDDIR)/%.nrm: $(SOURCEDIR)/%.sfd $(NORMALIZE)
-	@echo "  NORMALIZE  $(*F)"
-	@mkdir -p $(BUILDDIR)
-	@$(PY) $(NORMALIZE) $< $@
-	@if [ "`diff -u $< $@`" ]; then cp $@ $<; touch $@; fi
-
-$(BUILDDIR)/%.chk: $(SOURCEDIR)/%.sfd $(NORMALIZE)
-	@echo "  NORMALIZE  $(*F)"
-	@mkdir -p $(BUILDDIR)
-	@$(PY) $(NORMALIZE) $< $@
-	@diff -u $< $@ || (rm -rf $@ && false)
-
-%.svg: $(DOC)/%.tex $(OTF)
-	@echo "        SVG  $(*F)"
-	@xelatex --interaction=batchmode -output-directory=$(BUILDDIR) $< 1>/dev/null || (cat $(BUILDDIR)/$(*F).log && false)
-	@mutool draw -q -r 200 -o $@ $(BUILDDIR)/$(*F).pdf
-
-dist: check $(OTF) $(PDF) $(SVG)
-	@echo "       DIST  $(DIST).zip"
-	@rm -rf $(DIST) $(DIST).zip
-	@mkdir -p $(DIST)/$(DOC)
-	@cp $(OTF) $(DIST)
-	@cp $(PDF) $(SVG) $(DIST)/$(DOC)
-	@cp OFL.txt FONTLOG.txt AUTHORS.txt CONTRIBUTORS.txt $(DIST)
-	@cp README.md $(DIST)/README.txt
-	@cp CONTRIBUTING.md $(DIST)/CONTRIBUTING.txt
-	@zip -rq $(DIST).zip $(DIST)
-
-clean:
-	@rm -rf $(DIST) $(DIST).zip $(CHK) $(MIS) $(FEA) $(NRM) $(PDF) $(OTF)
+	cp $< $@
+
+$(BUILDDIR)/%.nrm: $(SOURCEDIR)/%.sfd $(NORMALIZE) | $(BUILDDIR)
+	$(info   NORMALIZE  $(*F))
+	$(PY) $(NORMALIZE) $< $@
+	if [ "`diff -u $< $@`" ]; then cp $@ $<; touch $@; fi
+
+$(BUILDDIR)/%.chk: $(SOURCEDIR)/%.sfd $(NORMALIZE) | $(BUILDDIR)
+	$(info   NORMALIZE  $(*F))
+	$(PY) $(NORMALIZE) $< $@
+	diff -u $< $@ || (rm -rf $@ && false)
+
+preview.svg: $(DOCSDIR)/preview.tex $(OTF) | $(BUILDDIR)
+	$(info         SVG  $@)
+	xelatex --interaction=batchmode \
+		-output-directory=$(BUILDDIR) \
+		$< 1> /dev/null || (cat $(BUILDDIR)/$(*F).log && false)
+
+preview.pdf: preview.svg
+	$(info         PDF  $@)
+	mutool draw -q -r 200 -o $< $@
+
+.PHONY: dist
+dist: check dist-clean $(OTF) $(PDF) $(SVG)
+	$(info         DIST  $(DIST).zip)
+	install -Dm644 $(OTF) -t $(DIST)
```

Comment:
The previous code was more explicit and easier to understand. How about using the full command options, like `--target-directory` and `--mode` instead of `-t` and `-m`?

Reply:
No. The `-t` flag is _very_ common in usage of `install` and far more common than its verbose brother. It's also the proper tool for the job, not `cp` in this case. I don't think explaining common shell syntax / usage is necessary in a context like this, that's what `man` pages are for.

This whole dist section is actually much _more_ explicit than it was before, and also more correct because the make targets have proper dependencies. I'm sorry make is a bear and kind of obtuse, but just because using it correctly makes it harder for you to follow doesn't make it worse. Phony targets that are just shell script placeholders may be "easier to understand" if what you understand is basic shell scripting, but that doesn't mean it's the best way to write a Makefile.

Reply:
I'm not sure if I came across as arguing for restoring the previous cp-based commands — I fully agree with using a proper tool like `install`, I was just suggesting using the flags that already come out of the box with it. Nothing about the new approach or its correctness would change; it just becomes more self-explanatory (which is helpful because, as you say yourself, make is not exactly the most intuitive to work with).

IIUC short options are most helpful when typing commands by hand, but in a script that's already written out, there's no downside in using the full option name, don't you agree?

Reply:
(If that's unacceptable to you, then at least consider splitting `-D` and `-m`, for readability, and adding a space after `-m` like you did with `-t`: `-D -m 644`.)

Reply:
First, `install` is not part of GNU Make, it is part of GNU's core utilities, which was in turn made to imitate the older BSD command of the same name. Tho long option names are a GNU only thing, they are not implemented in the BSD versions of make.

Secondly, even if the long options were universal (and in our case we are kind of assuming GNU tools anyway), there are several downsides to long arguments in scripts. One of them is that they tend to be so verbose they make simple lines that would otherwise be short enough to read so long that they wrap in the editor. This makes them _harder_ to read, not easier. Secondly the short options are (in this particular case) much more commonly used in the wild. You will see them in thousands of packaging scripts using these short options, and they are _more_ recognizable to anybody that works with these things than the long options. And less specific to this case, short options are also preferred in scripts where the CLI usage is part of many people's muscle memory. There is no sense in writing `tar xfz ...` on the CLI and `tar --extract --ungzip --file=...` in a script. Heck had to use `man` just to figure out what those are, and I would have to do the same for `install`, whereas thousands of times seeing and writing the short options have them drilled in my head and fingers.

Reply:
> then at least consider splitting -D and -m

Popular usage has these grouped together as `-Dm666`. I grepped the projects directory on my system (which has package repository sources files for several Linux distributions) and found several hundred thousand usages for `install -DmNNN` and just a few hundred with them split as in `install -D -m NNN`.

_YOU_ may not be familiar with the usage, but anybody with experience with `install` from other contexts is far more likely to have run into the combined usage, just like is is far mor common to see `tar xfz` than `tar -x -f -z` even though both are valid.

Reply:
Grouped short options are fine and very common, so I have no objection here.